editor/src/components/navigator/navigator-item/render-prop-selector-popup.tsx

```diff
@@ -85,35 +100,67 @@ export const RenderPropPicker = React.memo<RenderPropPickerProps>(({ key, id, ta
 
   const dispatch = useDispatch()
 
+  const projectContentsRef = useRefEditorState((state) => state.editor.projectContents)
+
   const onItemClick = React.useCallback(
-    (rawJSCodeForRenderProp: string) => (e: React.MouseEvent) => {
+    (preferredChildToInsert: PreferredChildToInsert) => (e: React.MouseEvent) => {
       e.stopPropagation()
       e.preventDefault()
 
+      const uid = generateConsistentUID(
+        'prop',
+        new Set(getAllUniqueUids(projectContentsRef.current).uniqueIDs),
+      )
+
+      const element = preferredChildToInsert.elementToInsert(uid)
+      if (element.type !== 'JSX_ELEMENT') {
+        throw new Error('only JSX elements are supported as preferred components')
```

Comment:
Is it sure? E.g. fragments could make sense or even expressions or just a string

Reply:
This specifically refers to elements coming from `preferredChildComponents`, and this is only valid until we add support for fragments and conditionals in render props (which we intend to do soon)